src/channel.js

```diff
@@ -1,46 +1,78 @@
 import {supabase} from './supabase-client.js'
+import {getUser} from './user.js'
 
-export const createChannel = async ({channel, user}) => {
-	const {name, slug} = channel
-	const {id: user_id} = user
+/**
+ * A channel
+ * @typedef {Object} Channel
+ * @property {string} name
+ * @property {string} slug - unique
+ * @property {string} [description]
+ */
+
+/**
+ * Creates a new radio channel and connects it to a user
+ * @param {Channel} fields
+ * @return {Promise<object>} {data, error}
+ */
+export const createChannel = async ({name, slug}) => {
+	const user = await getUser()
 
 	// Throw an error if the slug is in use by the old Firebase database.
 	const queryFirebaseDb = await firebaseGetChannelBySlug(slug)
 	const isSlugTaken = Object.keys(queryFirebaseDb).length > 0
-	if (isSlugTaken) throw new Error('Sorry. This channel slug is already taken by someone else.')
+	if (isSlugTaken) return {error: Error('Sorry. This channel slug is already taken by someone else.')}
```

Comment:
return `{error, code: 'channel-slug-taken-firebase'}` ?

Reply:
it's ncie with the codes. Like this? So we don't use the `Error` object.

```
const error = {message: 'blalbla', code: 'slug-taken'}
return {error}
```

or do we create an `Error` object and add the `code` prop? 🤔